sdk/core/azure-core/azure/core/exceptions.py

```diff
@@ -28,14 +28,35 @@
 import logging
 import sys
 
-from typing import Callable, Any, Optional, Union, Type, List, Dict, TYPE_CHECKING
+from typing import (
+    Callable,
+    Any,
+    Optional,
+    Union,
+    Type,
+    List,
+    Mapping,
+    TypeVar,
+    Generic,
+    Dict,
+    Protocol,
+    runtime_checkable,
+    TYPE_CHECKING,
+)
+from types import TracebackType
 
 _LOGGER = logging.getLogger(__name__)
 
 if TYPE_CHECKING:
-    from azure.core.pipeline.transport._base import _HttpResponseBase
     from azure.core.pipeline.policies import RequestHistory
 
+HTTPResponseType = TypeVar("HTTPResponseType")
+HTTPRequestType = TypeVar("HTTPRequestType")
+KeyType = TypeVar("KeyType")
```

Comment:
Key is status code which should be int?

Reply:
KeyType is used in a generic structure `ErrorMap` that don't need a specific type, the specifc type will be set by the instantiation.